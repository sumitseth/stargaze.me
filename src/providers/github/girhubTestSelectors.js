export const githubTestSelectors = {
  loginButton: 'app-login-button',
};
