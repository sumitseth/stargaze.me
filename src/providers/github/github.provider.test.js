import React from 'react';
import {
  render,
  getByTestId,
  fireEvent,
  waitForElement,
} from '@testing-library/react';
import renderer from 'react-test-renderer';

import GitHubClientProvider from './github.provider';
import { githubTestSelectors } from './girhubTestSelectors';

const fakeResponse = { token: '123' };
const fakeAuthenticator = jest.fn(() => Promise.resolve(fakeResponse));
const fakeClient = jest.fn(() => Promise.resolve(fakeResponse));

beforeEach(() => {
  fakeAuthenticator.mockClear();
  fakeClient.mockClear();
  jest.spyOn(window.localStorage.__proto__, 'setItem');
  jest.spyOn(window.localStorage.__proto__, 'getItem');
  window.localStorage.__proto__.setItem = jest.fn();
  window.localStorage.__proto__.getItem = jest.fn();
});

it('renders without crashing', () => {
  const { container } = render(
    <GitHubClientProvider authenticate={fakeAuthenticator} />
  );
  const login = getByTestId(container, githubTestSelectors.loginButton);
  expect(login.textContent).toBe('Sign In to Github');
});

it('authenticates', async () => {
  const { container } = render(
    <GitHubClientProvider authenticate={fakeAuthenticator} />
  );
  fireEvent.click(getByTestId(container, githubTestSelectors.loginButton));
  await waitForElement(
    () => getByTestId(container, githubTestSelectors.loginButton),
    { container }
  );
  const logout = getByTestId(container, githubTestSelectors.loginButton);
  expect(logout.textContent).toBe('Sign Out');
});

it('matches snapshot', () => {
  const tree = renderer
    .create(<GitHubClientProvider authenticate={fakeAuthenticator} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
