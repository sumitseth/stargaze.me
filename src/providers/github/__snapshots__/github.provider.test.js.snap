// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdVaJa sc-dnqmqq bUIycl"
  color="white"
>
  <nav>
    <a
      aria-current="page"
      className="sc-iwsKbI iSGNcw"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "red",
        }
      }
    >
      Home
    </a>
    <a
      className="sc-iwsKbI iSGNcw"
      href="/search"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
        }
      }
    >
      Search
    </a>
  </nav>
  <h1
    data-testid="app-title"
  >
    stargaze.me
  </h1>
  <a
    className="sc-bdVaJa sc-ifAKCX sc-htoDjs hSqoQA"
    color="blue"
    data-testid="app-login-button"
    href="#"
    onClick={[Function]}
  >
    Sign In to Github
  </a>
</div>
`;
